### PR TITLE
fix: gen-pkcs12-keystore adds ca.crt input option if it exists (#684)

### PR DESCRIPTION
**Pull request by smoldenhauer-ish**

### CODE REVIEW

**Comment by HoustonPutman** [comment]
Could you add some more information on what this fixes?

**Comment by smoldenhauer-ish** [comment]
I'll need to have a look into solrcloud_tls_test.go and see if I am able to understand it and can add a test in the next days.

File: controllers/util/solr_tls_util.go [hunk 1]
```diff
@@ -707,8 +707,11 @@ func (tls *TLSConfig) generatePkcs12InitContainer(imageName string, imagePullPol
 		},
 	}
 
-	cmd := "openssl pkcs12 -export -in " + DefaultKeyStorePath + "/" + TLSCertKey + " -in " + DefaultKeyStorePath +
-		"/ca.crt -inkey " + DefaultKeyStorePath + "/tls.key -out " + DefaultKeyStorePath +
+	caCrtFileName := DefaultKeyStorePath + "/ca.crt"
+
+	cmd := "OPTIONAL_CACRT=$(test -e " + caCrtFileName + " && echo ' -in " + caCrtFileName + "'); " +
```
Comment (HoustonPutman) [inline controllers/util/solr_tls_util.go:712]: One suggestion here is to put the `$(...)` inside of quotations, to make sure that the spaces are taken care of well. 
```suggestion
	cmd := "OPTIONAL_CACRT=\"$(test -e " + caCrtFileName + " && echo ' -in " + caCrtFileName + "')\"; " +
```

We would need to test it out to make sure that they are happy though

File: controllers/solrcloud_controller_tls_test.go [hunk 2]
```diff
@@ -522,7 +522,7 @@ func expectTLSConfigOnPodTemplateWithGomega(g Gomega, solrCloud *solrv1beta1.Sol
 				break
 			}
 		}
-		expCmd := "openssl pkcs12 -export -in /var/solr/tls/tls.crt -in /var/solr/tls/ca.crt -inkey /var/solr/tls/tls.key -out /var/solr/tls/pkcs12/keystore.p12 -passout pass:${SOLR_SSL_KEY_STORE_PASSWORD}"
+		expCmd := "OPTIONAL_CACRT=$(test -e /var/solr/tls/ca.crt && echo ' -in /var/solr/tls/ca.crt'); openssl pkcs12 -export -in /var/solr/tls/tls.crt $OPTIONAL_CACRT -inkey /var/solr/tls/tls.key -out /var/solr/tls/pkcs12/keystore.p12 -passout pass:${SOLR_SSL_KEY_STORE_PASSWORD}"
```
Comment (HoustonPutman) [inline controllers/solrcloud_controller_tls_test.go:525]:
```suggestion
		expCmd := "OPTIONAL_CACRT=\"$(test -e /var/solr/tls/ca.crt && echo ' -in /var/solr/tls/ca.crt')\"; openssl pkcs12 -export -in /var/solr/tls/tls.crt $OPTIONAL_CACRT -inkey /var/solr/tls/tls.key -out /var/solr/tls/pkcs12/keystore.p12 -passout pass:${SOLR_SSL_KEY_STORE_PASSWORD}"
```

**Comment by smoldenhauer-ish** [comment]
Hope this is good for merge, now. 
BTW, I'm offline for the next two weeks.

File: helm/solr-operator/Chart.yaml [hunk 3]
```diff
@@ -54,13 +54,21 @@ annotations:
   # Add change log for a single release here.
   # Allowed syntax is described at: https://artifacthub.io/docs/topics/annotations/helm/#example
   artifacthub.io/changes: |
+    - kind: fixed
+      description: gen-pkcs12-keystore init container fails if the tls secret contains no ca.crt
```
Comment (gerlowskija) [inline helm/solr-operator/Chart.yaml:58]:
```suggestion
      description: gen-pkcs12-keystore initContainer now supports 'ca.crt'-less TLS secrets
```

Most of the other 'descriptions' here highlight the correct/improved behavior, so I've suggested a similar change above to stay in line with that.

**Comment by gerlowskija** [comment]
Hi @smoldenhauer-ish - this LGTM overall.

I did try to push one minor change to your branch, to reword the `helm/solr-operator/Chart.yaml` entry slightly.  But it looks like I don't have the requisite permissions to collaborate on that branch.  If you're willing to share write-access on the PR branch, I'll try sharing again?  (Or alternately, you could incorporate the ["Suggested Change" here](https://github.com/apache/solr-operator/pull/685/files#r1876310454).)

Otherwise this is ready to merge IMO.  I'll hold off for a bit in hopes you see this and answer, but won't block merging the PR too long, as it's just a minor doc tweak.

**Comment by smoldenhauer-ish** [comment]
Hi @gerlowskija - added your suggestion.

**Review by gerlowskija — APPROVED** [review]

**Pull request merged**